### DISCUSSION TITLE
Exporting a store yields a typescript error when 'composite: true'

### DISCUSSION THREAD

**Discussion by onionhammer** [pmndrs/jotai#2725] [Bug report]
### Bug Description

![image](https://github.com/user-attachments/assets/0446c608-6182-4a0f-8584-cc35737856ed)

Exported variable 'formStore' has or is using name 'PrdStore' from external module "node_modules/jotai/esm/vanilla/store" but cannot be named.(4023)

### Reproduction Link

https://stackblitz.com/edit/vitejs-vite-umx5x6?file=src%2Fglobals.ts

**Comment by dai-shi** [comment]
Related: #2717

I didn't know this happens only with composite:true. Do you have declaration:true?

**Reply by onionhammer** [reply]
Maybe I'm not following, but this error is in a client-app, not the library.     "declaration": true doesnt seem to make a difference to the client app.

**Reply by dai-shi** [reply]
Yeah, it seems only adding "composite:true" makes the error.
Our exported types are intentional, so can you please do the workaround for now?

https://stackblitz.com/edit/vitejs-vite-bjw6hr?file=src%2Fglobals.ts
